### PR TITLE
Specialization/attribute/skill randomization in class creation widget

### PR DESCRIPTION
**Pull request by wareya**
The button text is defined in the layout file (like it is for a bunch of the settings menus) since I have no idea where else to put it.

### CODE REVIEW

**Comment by wareya** [comment]
Note: the way random skills are selected is still buggy, I just wanted to get the PR up here.

image

Note Short Blade being in both columns.

**Comment by wareya** [comment]
Alright that should fix it.

File: apps/openmw/mwgui/class.cpp [hunk 1]
```diff
@@ -12,8 +12,13 @@
 
 #include "../mwworld/esmstore.hpp"
 
+#include "../../../components/misc/rng.hpp"
```
Comment (Capostrophic) [inline apps/openmw/mwgui/class.cpp]: Why not include it like ``<components/misc/rng.hpp>`` like anything else does?

File: files/mygui/openmw_chargen_create_class.layout [hunk 2]
```diff
@@ -76,6 +76,9 @@
         <!-- Dialog buttons -->
         <Widget type="HBox" position="0 158 482 24">
             <Widget type="Spacer"/>
+            <Widget type="AutoSizedButton" skin="MW_Button" name="RandomButton">
+                <Property key="Caption" value="Random"/>
```
Comment (Capostrophic) [inline files/mygui/openmw_chargen_create_class.layout]: Non-localizable strings are not good.

File: apps/openmw/mwgui/class.cpp (hunk 1 above)
Comment (wareya) [reply apps/openmw/mwgui/class.cpp]: Oh, I didn't know the include paths were like that. I'll do that.

File: files/mygui/openmw_chargen_create_class.layout (hunk 2 above)
Comment (wareya) [reply files/mygui/openmw_chargen_create_class.layout]: Where are openmw-specific localizable strings stored?

Comment (akortunov) [reply files/mygui/openmw_chargen_create_class.layout]: Nowhere. Zini said that custom localization is only post-1.0.

**Comment by zinnschlag** [comment]
Not in favour of this change (pre 1.0). That's just more stuff we have to fix up once the de-hardcoding is under way. But a neat idea for post-1.0.

**Comment by zinnschlag** [comment]
I am going to close this PR. By the time we are ready to merge, OpenMW will have changed so much that most of the code will have to be re-written anyway.

**Pull request closed without merging**